pkg/kubelet/kubelet_test.go

```diff
@@ -455,15 +457,7 @@ func TestSyncPodsWithTerminationLog(t *testing.T) {
 
 // drainWorkers waits until all workers are done.  Should only used for testing.
 func (kl *Kubelet) drainWorkers() {
-	for {
-		kl.podWorkers.lock.Lock()
-		length := len(kl.podWorkers.workers)
-		kl.podWorkers.lock.Unlock()
-		if length == 0 {
-			return
-		}
-		time.Sleep(time.Millisecond * 100)
-	}
+	kl.podWorkers.waitGroup.Wait()
```

Comment:
I slightly changed your suggestion, but I agree that it's much cleaner when there is no test code in podWorkers class. Hopefully you will also like it more.